src/render.rs

```diff
@@ -372,3 +379,101 @@ fn message_link(config: &Config, event_id: &EventId) -> String {
         room_id, event_id
     )
 }
+
+#[cfg(feature = "ping-rooms")]
+#[derive(serde::Serialize, serde::Deserialize, Clone, Debug)]
+struct PingJsonResponse {
+    pub disclaimer: String,
+    pub pings: std::collections::BTreeMap<String, PingJsonResponsePing>,
+    pub mean: f32,
+    pub pongservers: std::collections::BTreeSet<String>,
+}
+
+#[cfg(feature = "ping-rooms")]
+#[derive(serde::Serialize, serde::Deserialize, Clone, Debug)]
+struct PingJsonResponsePing {
+    pub pings: std::collections::BTreeSet<String>,
+    pub pongs: std::collections::BTreeMap<String, PingJsonResponsePong>,
+    pub mean: f32,
+    pub median: f32,
+    pub gmean: f32,
+}
+
+#[cfg(feature = "ping-rooms")]
+#[derive(serde::Serialize, serde::Deserialize, Clone, Debug)]
+struct PingJsonResponsePong {
+    pub diffs: std::collections::BTreeMap<String, u64>,
+    pub mean: f32,
+    pub median: f32,
+    pub gmean: f32,
+}
+
+#[cfg(feature = "ping-rooms")]
+async fn render_ping_statistics(config: Config) -> String {
+    // Prepare header of the section
+    let mut output =String::from( "## Dept of Ping 🏓
+    
+Here we reveal, rank, and applaud the homeservers with the lowest ping, as measured by [pingbot](https://github.com/maubot/echo), a [maubot](https://github.com/maubot/maubot) that you can host on your own server.\n\n");
```

Comment:
This indentation is important as rust strings preserve the spaces sadly and don't remove the function's indentation from them